### PR TITLE
fix:setStatus for WalletConnect connector on syncAccount

### PR DESCRIPTION
**Pull request by KannuSingh**
# Description

setStatus for WalletConnect connector on syncAccount

## Type of change

- [ ] Chore (non-breaking change that addresses non-functional tasks, maintenance, or code quality improvements)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Associated Issues

For Linear issues: Closes APKT-xxx
For GH issues: closes #...

# Showcase (Optional)

If there is a UI change include the screenshots with before and after state.
If new feature is being introduced, include the link to demo recording.

# Checklist

- [x] Code in this PR is covered by automated tests (Unit tests, E2E tests)
- [x] My changes generate no new warnings
- [x] I have reviewed my own code
- [x] I have filled out all required sections
- [ ] I have tested my changes on the preview link
- [ ] Approver of this PR confirms that the changes are tested on the preview link

### CODE REVIEW

File: packages/adapters/wagmi/src/client.ts [hunk 1]
```diff
@@ -682,7 +682,12 @@ export class WagmiAdapter implements ChainAdapter {
 
     if (this.wagmiConfig) {
       if (connector) {
-        if (connector.name === 'WalletConnect' && connector.getProvider && address) {
+        if (
+          status === 'connected' &&
```
Comment (tomiir) [inline packages/adapters/wagmi/src/client.ts:686]: @enesozturk @svenvoskamp why weren't we checking if this was connected? 

Might be a reason for this from the multichain side?

File: packages/adapters/wagmi/src/client.ts [hunk 2]
```diff
@@ -701,6 +705,7 @@ export class WagmiAdapter implements ChainAdapter {
 
             this.appKit?.setPreferredAccountType(preferredAccountType, chainNamespace)
             this.appKit?.setCaipAddress(caipAddress, chainNamespace)
+            this.appKit?.setStatus('connected', chainNamespace)
```
Comment (tomiir) [inline packages/adapters/wagmi/src/client.ts:708]: Maybe not check for connected and set the state directly?

Comment (KannuSingh) [reply packages/adapters/wagmi/src/client.ts]: Done.

**Review by enesozturk — APPROVED** [review]

File: packages/adapters/wagmi/src/client.ts [hunk 3]
```diff
@@ -701,6 +700,7 @@ export class WagmiAdapter implements ChainAdapter {
 
             this.appKit?.setPreferredAccountType(preferredAccountType, chainNamespace)
             this.appKit?.setCaipAddress(caipAddress, chainNamespace)
+            this.appKit?.setStatus(status, chainNamespace)
```
Comment (enesozturk) [inline packages/adapters/wagmi/src/client.ts:703]: @KannuSingh I guess we might need to do same for disconnect status, could you do that as well?

Comment (KannuSingh) [reply packages/adapters/wagmi/src/client.ts:703]: okay, I will check and do the same.

**Pull request merged**